storm-core/src/jvm/org/apache/storm/utils/ConfigUtils.java

```diff
@@ -0,0 +1,678 @@
+            dir = String.valueOf(conf.get("storm.log.dir"));
+        } else {
+            if (System.getProperty("storm.home") != null) {
+                dir = System.getProperty("storm.home") + FILE_SEPARATOR + "logs";
+            } else {
+                dir = FILE_SEPARATOR + "logs";
+            }
+        }
+        try {
+            return new File(dir).getCanonicalPath();
+        } catch (IOException ex) {
+            throw new IllegalArgumentException("Illegal storm.log.dir in conf: " + dir);
+        }
+    }
+
+    public static String clojureConfigName(String name) {
+        return name.toUpperCase().replace("_", "-");
+    }
+
+    // ALL-CONFIGS is only used by executor.clj once, do we want to do it here? TODO
+    public static List<Object> All_CONFIGS() {
+        List<Object> ret = new ArrayList<Object>();
+        Config config = new Config();
+        Class<?> ConfigClass = config.getClass();
+        Field[] fields = ConfigClass.getFields();
+        for (int i = 0; i < fields.length; i++) {
+            try {
+                Object obj = fields[i].get(null);
+                ret.add(obj);
+            } catch (IllegalArgumentException e) {
+                LOG.error(e.getMessage(), e);
+            } catch (IllegalAccessException e) {
+                LOG.error(e.getMessage(), e);
+            }
+        }
+        return ret;
+    }
+
+    public static String clusterMode(Map conf) {
+        String mode = (String)conf.get(Config.STORM_CLUSTER_MODE);
+        return mode;
+    }
+
+    public static boolean isLocalMode(Map conf) {
+        String mode = (String)conf.get(Config.STORM_CLUSTER_MODE);
+        if (mode != null) {
+            if ("local".equals(mode)) {
+                return true;
+            }
+            if ("distributed".equals(mode)) {
+                return false;
+            }
+        }
+        throw new IllegalArgumentException("Illegal cluster mode in conf: " + mode);
+    }
+
+    public static int samplingRate(Map conf) {
+        double rate = Utils.getDouble(conf.get(Config.TOPOLOGY_STATS_SAMPLE_RATE));
+        if (rate != 0) {
+            return (int) (1 / rate);
+        }
+        throw new IllegalArgumentException("Illegal topology.stats.sample.rate in conf: " + rate);
+    }
+
+    // public static mkStatsSampler // depends on Utils.evenSampler() TODO, this is sth we need to do after util
+    // public static readDefaultConfig // depends on Utils.clojurifyStructure and Utils.readDefaultConfig // TODO
+    // validate-configs-with-schemas is just a wrapper of ConfigValidation.validateFields(conf)
+
+    //For testing only
+    // for java
+    // try (SetMockedStormConfig mocked = new SetMockedStormConfig(conf)) {
+    //    run test ...
+    // }
+    //
+    // for clojure
+    // (with-open [mock (SetMockedStormConfig. conf)]
+    //     run test ...)
+    public static class SetMockedStormConfig implements Closeable {
```

Comment:
Should we maybe call this MockedStormConfigGuard or something with 'guard' in it? It would be nice to directly indicate that this object is an RAII-style guard.